sqlmesh/core/plan/builder.py

```diff
@@ -673,6 +719,95 @@ def _ensure_new_env_with_changes(self) -> None:
+
+            def _raise_or_warn() -> None:
+                if s_id_snapshot.model.on_destructive_change.is_error:
+                    raise PlanError(error_msg)
+                logger.warning(warning_msg)
+
+            subdag = set([s_id, *upstream_snapshot_ids])
+            subdag_to_process = subdag.copy()
+            subdag_no_cols_to_types = set()
+
+            # get already classified snapshots
+            for id in subdag:
+                if id.name in self._snapshot_change_is_destructive:
+                    subdag_to_process.remove(id)
+                    value = self._snapshot_change_is_destructive[id.name]
+                    if value:
+                        _raise_or_warn()
+                        return
+                    if value is None:
+                        subdag_no_cols_to_types.add(id)
+
+            # process unclassified snapshots
+            for id in subdag_to_process:
+                snapshot = t.cast(
+                    Snapshot,
+                    s_id_snapshot if id == s_id else self._context_diff.snapshots.get(id),
+                )
+                new, old = self._context_diff.modified_snapshots[snapshot.name]
+
+                # we must know all columns_to_types to determine whether a change is destructive
+                old_columns_to_types = old.model.columns_to_types
+                new_columns_to_types = new.model.columns_to_types
+                if (
+                    not old_columns_to_types
+                    or not columns_to_types_all_known(old_columns_to_types)
+                    or not new_columns_to_types
+                    or not columns_to_types_all_known(new_columns_to_types)
+                ):
+                    self._snapshot_change_is_destructive[id.name] = None
+                    subdag_no_cols_to_types.add(id)
+                    continue
+
+                schema_differ = self._engine_schema_differ
+                schema_diff = schema_differ.compare_columns(
+                    new.model.name,
+                    t.cast(t.Dict[str, exp.DataType], old_columns_to_types),
+                    t.cast(t.Dict[str, exp.DataType], new_columns_to_types),
+                )
+
+                has_drop = any(
+                    [
+                        isinstance(action, exp.Drop)
+                        for actions in schema_diff
+                        for action in actions.args.get("actions", [])
+                    ]
+                )
+                self._snapshot_change_is_destructive[id.name] = has_drop
+                if has_drop:
+                    _raise_or_warn()
+                    return
+
+            if subdag_no_cols_to_types:
+                logger.info(
+                    f"Unable to determine at plan time if changes cause a destructive schema change to model '{s_id.name}'."
```

Comment:
why is it unable to determine it? (because we don't have the types, how can they avoid this warning in the future)

Reply:
This is not a warning though

Reply:
then what is it?